web_widget_html_markdown/static/src/js/web_widget_html_markdown.js

```diff
@@ -0,0 +1,426 @@
+            LIBS_PATH + "bootstrap-markdown.js",
+        ],
+        cssLibs: [
+            LIBS_PATH + "bootstrap-markdown.min.css",
+            CUST_LIBS_PATH + "web_widget_html_markdown.css",
+        ],
+
+        
+        /*=========================INIT AND START===============*/
+        init: function (parent, options) {
+            this._super.apply(this, arguments);
+            this.isSwitch = false;
+            this.options = _.extend({
+                default_markdown_on_new: true,
+                only_markdown_on_new : true,
+            }, options || {});
+        },
+        start: function() {
+            this._super();
+            // Inject MARKDOWN BUTTON
+            var md_s = $
+        (
+" <div id='radio_container'> <input id='markdown' type='radio' name='mk_html_switcher' value='markdown'>"+
+" <label id='markdown_label' for='markdown'>Markdown</label>  " +
+"<input type='radio' id='html' name='mk_html_switcher' value='html' checked >" +
+"  <label id='html_lable' for='html'>HTML</label></div><div id='md_infoarea'></div>")
+            var self = this;
+            this.$el.prepend(md_s);
+            //
+            this.$el.find('input').addClass('mk_html_switch');
+            this.$el.find('input').change([self], self._switch_markdown_html);
+            // widgets elements as general object attributes, used everywhere
+            this.infoarea = this.$el.find('div#md_infoarea');
+            this.input_markdown = this.$el.find('input#markdown');
+            this.input_html = this.$el.find('input#html');
+            this.markdown_label = this.$el.find('label#markdown_label');
+            this.html_label = this.$el.find('label#html_label');
+        },
+
+        /*============= table formatting helper ========================= */
+
+        _injectTableFormatting: function() {
+           this.$content.find('table').addClass('table');
+           this.$content.find('table').addClass('table-bordered');
+           this.$content.find('table').addClass('table-striped');
+           this.$content.find('table > thead').addClass('thead-light');
+        },
+
+
+        /*=========================HTML WIDGET OVERIDES AND EXTENSIONS===============*/
+        commitChanges: function () {
+            var self = this;
+            if (config.isDebug() && this.mode === 'edit') {
+                var layoutInfo = $.summernote.core.dom.makeLayoutInfo(this.wysiwyg.$editor);
+                $.summernote.pluginEvents.codeview(undefined, undefined, layoutInfo, false);
+            }
+             if (self.mode == 'edit' && self.$is_markdown) {
+                 // Transform in html before saving
+                 return self.sw_mk_to_html().then(function() {
+                     self.$is_markdown = false;
+                     // we use recursion (max1) is_markdown is reset
+                    return self.commitChanges()
+                 });
+            } else if (self.mode == 'edit') {
```

Comment:
No/ this is taken from source HTML. if we are not in edit mode commit changes shouldnt be ran.
We have replicated 5 lines of code because we had to add forceChange:true.
I explain in a brief comment.

Not done.